sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/data/SampleTrackingConnector.java

```diff
@@ -0,0 +1,305 @@
+package life.qbic.portal.sampletracking.data;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import life.qbic.business.download.DownloadManifest;
+import life.qbic.business.download.DownloadManifestFormatter;
+import life.qbic.portal.sampletracking.view.projects.viewmodel.ProjectStatus;
+import org.apache.hc.client5.http.auth.AuthScope;
+import org.apache.hc.client5.http.auth.UsernamePasswordCredentials;
+import org.apache.hc.client5.http.classic.methods.HttpGet;
+import org.apache.hc.client5.http.impl.auth.BasicCredentialsProvider;
+import org.apache.hc.client5.http.impl.classic.CloseableHttpClient;
+import org.apache.hc.client5.http.impl.classic.CloseableHttpResponse;
+import org.apache.hc.client5.http.impl.classic.HttpClients;
+import org.apache.hc.core5.http.HttpEntity;
+import org.apache.hc.core5.http.ParseException;
+import org.apache.hc.core5.http.io.entity.EntityUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * <b>short description</b>
+ *
+ * <p>detailed description</p>
+ *
+ * @since <version tag>
+ */
+public class SampleTrackingConnector implements ProjectStatusProvider, SampleStatusProvider, DownloadManifestProvider {
+
+  private static final Logger log = LoggerFactory.getLogger(SampleTrackingConnector.class);
+
+  public final String samplesSuffix;
+  public final String statusSuffix;
+  public final String projectsSuffix;
```

Comment:
NIT: was not clear to me that these are URL endpoints or parts of them, would be clearer if renamed

Reply:
What names would be clearer here? I had difficulties finding expressive variable names so I welcome any suggestions 😄